### PR TITLE
Use fresh `ReadView` for mutate queries

### PR DESCRIPTION
**Pull request by xgreenx**
When we manually produce blocks or `submit_and_wait` for the transaction, the `ReadView` has a view of the database at height X.

But when we return a response, it needs to use the view at height X + 1.

The change uses the latest view for mutate queries to solve this problem.

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

**Comment by Dentosal** [comment]
Lots of conflicts with `master`.

**Comment by xgreenx** [comment]
Fixed=)

File: crates/fuel-core/src/schema/chain.rs [hunk 1]
```diff
@@ -816,20 +814,18 @@ impl ChainInfo {
...
 
     async fn da_height(&self, ctx: &Context<'_>) -> U64 {
-        let query: &ReadView = ctx.data_unchecked();
-
-        let height = query
-            .da_height()
-            .expect("The blockchain always should have genesis block");
+        let Ok(query) = ctx.read_view() else {
+            return 0.into();
+        };
```
Comment (Dentosal) [inline crates/fuel-core/src/schema/chain.rs:826]: Is this change of behavior intentional? Is returning zero correct?

Comment (xgreenx) [reply crates/fuel-core/src/schema/chain.rs:826]: If I change the output type to `Result`, it will change the schema, and it will be a breaking change.

Comment (Dentosal) [reply crates/fuel-core/src/schema/chain.rs:826]: My question is: should we rather panic or return zero? But all things considered I think zero is fine here.

**Review by Dentosal — APPROVED** [review]

**Pull request merged**